src/test/shader.js

```diff
@@ -0,0 +1,640 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+/* eslint-disable no-var, max-statements */
+import 'babel-polyfill';
+import {document, window} from 'global';
+import {Buffer, createGLContext, Program, glGetDebugInfo} from 'luma.gl';
+
+const glslify = require('glslify');
+
+// Utilities functions that to be moved to a common place for future tests
+
+function glEnumToString(gl, value) {
+  // Optimization for the most common enum:
+  if (value === gl.NO_ERROR) {
+    return "NO_ERROR";
+  }
+  for (var p in gl) {
+    if (gl[p] == value) {
+      return p;
+    }
+  }
+  return "0x" + value.toString(16);
+};
+
+function addSpan(contents, div) {
```

Comment:
While it is cool to have an HTML interface, it should be simpler/less maintenance to just use `tape` to run the tests and have `tap-browser-color` HTML-ify the resulting TAP results. This is already set up in the CI tests. We could just have a folder for tests that should only run under the browser, if that is an issue.